core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1842,7 +1870,7 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
                 // Ignore the case where a tx appears on a side chain at the same time as the best chain (this is
                 // quite normal and expected).
                 Sha256Hash hash = tx.getHash();
-                if (!unspent.containsKey(hash) && !spent.containsKey(hash)) {
+                if (!unspent.containsKey(hash) && !spent.containsKey(hash) && !dead.containsKey(hash)) {
```

Comment:
That is exactly what this new check is doing. It checks a tx is NOT dead before calling commitTx(tx). If tx was dead, it will remain dead.